src/components/ErrorBoundary.tsx

```diff
@@ -0,0 +1,48 @@
+import { Component, ErrorInfo, ReactNode } from 'react';
+
+interface Props {
+  children: ReactNode;
+  fallback?: ReactNode;
+}
+
+interface State {
+  hasError: boolean;
+  error?: Error;
+}
+
+class ErrorBoundary extends Component<Props, State> {
```

Comment:
Unfortunately, there's still no way to implement ErrorBoundary component using react's functional components. So class may be ok.
Another otion - use some third-party libraries like [eact-error-boundary](https://www.npmjs.com/package/react-error-boundary)

No need to use it here. Just a hint